crates/red_knot_python_semantic/resources/mdtest/unpacking.md

```diff
@@ -306,3 +306,110 @@ reveal_type(b)  # revealed: Unknown
 reveal_type(a)  # revealed: LiteralString
 reveal_type(b)  # revealed: LiteralString
 ```
+
+## Union
```

Comment:
We should probably have some tests of what happens when one element of the union is the wrong length?

And some tests for unions like `tuple[int, str] | IterableOfBytes()` where `IterableOfBytes` is a type that defines `__iter__` etc so that iterating it returns some type.

And maybe some tests for nested unpacking involving unions (I mean like where we have `a, (b, c) = ...` and unions on the RHS.)

Unpacking unions when there are starred elements on the LHS?

Reply:
Added these test cases:
* Nested tuple types
* Starred expressions
* Tuple size mismatch
* Union of custom iterator with a tuple type